src/commonMain/kotlin/com/benasher44/uuid/uuid.kt

```diff
@@ -181,3 +185,35 @@ public class Uuid(val uuid: ByteArray = genUuid()) {
      */
     override fun hashCode(): Int = uuid.contentHashCode()
 }
+
+/**
+ * Set the [Uuid.version] on this big-endian [ByteArray]. The [Uuid.variant] is
+ * always set to the RFC 4122 one since this is the only variant supported by
+ * the [Uuid] implementation.
+ *
+ * @return Itself after setting the [Uuid.variant] and [Uuid.version].
+ */
+@Suppress("NOTHING_TO_INLINE")
+// @kotlin.internal.InlineOnly
+internal inline fun ByteArray.setVersion(version: Int) = apply {
+    this[6] = ((this[6].toInt() and 0x0F) or (version shl 4)).toByte()
+    this[8] = ((this[8].toInt() and 0x3F) or 0x80).toByte()
+}
+
+/**
+ * Construct new version 4 [UUID][Uuid].
+ *
+ * Version 4 UUIDs are randomly generated from the best available random source.
+ * The selection of that source is depends on the platform. Some systems may be
+ * bad at generating sufficient entropy, e.g. virtual machines, and this might
+ * lead to collisions faster than desired. Version 5 may be used if this is the
+ * case and no other measures are possible to increase the entropy of the random
+ * source of the platform.
+ *
+ * @return New version 4 [UUID][Uuid] of random data.
+ * @sample UuidTest.uuid4_generation
```

Comment:
Whoops looks like the find/replace got one too many :)

Reply:
The name is correct but `UuidTest` does not reside in a package and I didn't use auto-import from IntelliJ because the current EAP has huge problems…